spark/v3.1/spark/src/test/java/com/netease/arctic/spark/TestDropPartitions.java

```diff
@@ -0,0 +1,81 @@
+package com.netease.arctic.spark;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestDropPartitions extends SparkTestBase {
+  private final String database = "db_arctic";
+
+  @Before
+  public void prepare() {
+    sql("use " + catalogNameArctic);
+    sql("create database if not exists " + database);
+  }
+
+  @After
+  public void cleanUpTable() {
+    sql("drop database " + database);
+  }
+
+  @Test
+  public void testAlterKeyedTableDropPartitions() {
+    String dropPartitionTable = "drop_partition_table";
+    sql("create table {0}.{1} ( \n" +
+        " id int , \n" +
+        " name string , \n " +
+        " ts int, primary key(id) " +
+        ") using arctic \n" +
+        " partitioned by (ts, name) \n" +
+        " tblproperties ( \n" +
+        " ''props.test1'' = ''val1'', \n" +
+        " ''props.test2'' = ''val2'' ) ", database, dropPartitionTable);
+    sql("insert overwrite {0}.{1} "+
+        " values (1, ''aaa'', 1 ) , " +
+        "(4, ''bbb'', 2), " +
+        "(5, ''ccc'', 3) ", database, dropPartitionTable);
+
+    sql("insert into {0}.{1} "+
+        " values (2, ''aaa'', 1 ) , " +
+        "(3, ''bbb'', 2), " +
+        "(6, ''ccc'', 3) ", database, dropPartitionTable);
+    rows = sql("select * from {0}.{1}", database,dropPartitionTable);
+    Assert.assertEquals(6, rows.size());
+
+    sql("alter table {0}.{1} drop if exists partition (ts=1, name=''aaa'')", database, dropPartitionTable);
+    sql("select * from {0}.{1}", database,dropPartitionTable);
+    Assert.assertEquals(4, rows.size());
+    sql("drop table {0}.{1}", database, dropPartitionTable);
+  }
+
+  @Test
+  public void testAlterUnKeyedTableDropPartitions() {
+    String dropPartitionTable = "drop_partition_table";
+    sql("create table {0}.{1} ( \n" +
+        " id int , \n" +
+        " name string , \n " +
+        " ts int" +
+        ") using arctic \n" +
+        " partitioned by (ts, name) \n" +
+        " tblproperties ( \n" +
+        " ''props.test1'' = ''val1'', \n" +
+        " ''props.test2'' = ''val2'' ) ", database, dropPartitionTable);
+    sql("insert overwrite {0}.{1} "+
+        " values (1, ''aaa'', 1 ) , " +
+        "(4, ''bbb'', 2), " +
+        "(5, ''ccc'', 3) ", database, dropPartitionTable);
+
+    sql("insert into {0}.{1} "+
+        " values (2, ''aaa'', 1 ) , " +
+        "(3, ''bbb'', 2), " +
+        "(6, ''ccc'', 3) ", database, dropPartitionTable);
+    rows = sql("select * from {0}.{1}", database,dropPartitionTable);
+    Assert.assertEquals(6, rows.size());
+
+    sql("alter table {0}.{1} drop if exists partition (ts=1, name=''aaa'')", database, dropPartitionTable);
+    sql("select * from {0}.{1}", database,dropPartitionTable);
+    Assert.assertEquals(4, rows.size());
+    sql("drop table {0}.{1}", database, dropPartitionTable);
+  }
```

Comment:
The test case should include 
- Keyed/Un-keyd
- Hive/Non-Hive 

and in each case should check:

1. Data is right, the MOR result will not include dropped partition data 
2. Metadata is right,  the HMS Partition is deleted for HiveTable
3. Make sure no change file is created.